### PR TITLE
Add call to installer-gather.sh on failure

### PR DESCRIPTION
**Pull request by sdodson**
Don't remove existing artifact gathering just yet.

Depends on https://github.com/openshift/installer/pull/1561

### CODE REVIEW

File: ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml [hunk 1]
```diff
@@ -349,6 +349,9 @@ objects:
                         --key /tmp/artifacts/installer/tls/journal-gatewayd.key \
                         --url "https://${bootstrap_ip}:19531/entries?_SYSTEMD_UNIT=${service}.service"
                 done
+                ssh-add /tmp/cluster/ssh-privatekey
+                ssh -A core@${bootstrap_ip} /usr/local/bin/installer-gather.sh
```
Comment (sferich888) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml]: Can you do this in 1 command with `-i /tmp/cluster/ssh-privatekey` added to this command (and the one below it)?

**Comment by deads2k** [comment]
This will be awesome!

@openshift/sig-master

File: ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml (hunk 1 above)
Comment (sdodson) [reply ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml]: No, that doesn't appear to forward the credential properly.

**Comment by wking** [comment]
Why is this WIP?

[MCO-e2e-aws][1]:

```
level=fatal msg="failed to initialize the cluster: Cluster operator marketplace is still updating: timed out waiting for the condition"
```

[ovn-e2e-aws][2]:

```
level=fatal msg="failed to wait for bootstrap-complete event: timed out waiting for the condition"
```

It looks like there's an issue with the SSH-agent:

```console
$ curl -s https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/3475/rehearse-3475-pull-ci-openshift-ovn-kubernetes-master-e2e-aws/1/artifacts/e2e-aws/container-logs/teardown.log.gz | gunzip | grep -3 'Could not open a connection to your authentication agent'
curl --insecure --silent --connect-timeout 5 --retry 3 --cert /tmp/artifacts/installer/tls/journal-gatewayd.crt --key /tmp/artifacts/installer/tls/journal-gatewayd.key --url https://3.84.112.162:19531/entries?_SYSTEMD_UNIT=openshift.service
curl --insecure --silent --connect-timeout 5 --retry 3 --cert /tmp/artifacts/installer/tls/journal-gatewayd.crt --key /tmp/artifacts/installer/tls/journal-gatewayd.key --url https://3.84.112.162:19531/entries?_SYSTEMD_UNIT=kubelet.service
curl --insecure --silent --connect-timeout 5 --retry 3 --cert /tmp/artifacts/installer/tls/journal-gatewayd.crt --key /tmp/artifacts/installer/tls/journal-gatewayd.key --url https://3.84.112.162:19531/entries?_SYSTEMD_UNIT=crio.service
Could not open a connection to your authentication agent.
No user exists for uid 1263960000
unknown user 1263960000
oc --insecure-skip-tls-verify --request-timeout=5s get apiserver.config.openshift.io authentication.config.openshift.io build.config.openshift.io console.config.openshift.io dns.config.openshift.io featuregate.config.openshift.io image.config.openshift.io infrastructure.config.openshift.io ingress.config.openshift.io network.config.openshift.io oauth.config.openshift.io project.config.openshift.io scheduler.config.openshift.io -o json
```

[1]: https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_release/3475/rehearse-3475-pull-ci-openshift-machine-config-operator-master-e2e-aws/1
[2]: https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_release/3475/rehearse-3475-pull-ci-openshift-ovn-kubernetes-master-e2e-aws/1

**Comment by sdodson** [comment]
> Why is this WIP?

I was trying to figure out if it worked or not. I'll have to look into how to fix the agent.

**Comment by sdodson** [comment]
Last rehearsal didn't seem to include the teardown from this job. I thought it would only select jobs that would be affected by the change?

**Comment by abhinavdahiya** [comment]
@sdodson 
I think you might also wanna add this to 
https://github.com/openshift/release/blob/master/ci-operator/templates/openshift/installer/cluster-launch-installer-src.yaml
https://github.com/openshift/release/blob/master/ci-operator/templates/openshift/openshift-ansible/cluster-launch-e2e-40.yaml

**Comment by sdodson** [comment]
I'm relatively certain that the items required to find the bootstrap ip are also there in the ansible job but I at least want to get proof that this works in general before I add it there as well. 

roll 1d100 and hope for the best...

File: ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml [hunk 2]
```diff
@@ -349,6 +349,14 @@ objects:
                         --key /tmp/artifacts/installer/tls/journal-gatewayd.key \
                         --url "https://${bootstrap_ip}:19531/entries?_SYSTEMD_UNIT=${service}.service"
                 done
+                if ! whoami &> /dev/null; then
+                  if [ -w /etc/passwd ]; then
```
Comment (abhinavdahiya) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml]: this is read-only file

File: ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml [hunk 3]
```diff
@@ -349,6 +349,14 @@ objects:
...
+                    echo "${USER_NAME:-default}:x:$(id -u):0:${USER_NAME:-default} user:${HOME}:/sbin/nologin" >> /etc/passwd
+                  fi
+                fi
+                ssh-add /tmp/cluster/ssh-privatekey
```
Comment (abhinavdahiya) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml]: the key exists at `/etc/openshift-installer/ssh-privatekey`

Comment (abhinavdahiya) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml]: also you need to run `eval $(ssh-agent)` to run the agent and setup the env

**Comment by sdodson** [comment]
depends on https://github.com/openshift/origin/pull/22592

File: ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml (hunk 3 above)
Comment (sferich888) [reply ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml]: Do we even need to use the agent? 
Can you use `-i` to pass in the key?

Comment (wking) [reply ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml]: > Can you use `-i` to pass in the key?

That covers the CI -> bootstrap hop, but you need the agent for subsequent bootstrap -> control-plane hops.

File: ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml (hunk 1 above)
Comment (sferich888) [reply ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml]: Interesting.

File: ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml (hunk 3 above)
Comment (sferich888) [reply ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml]: I see.

File: ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml [hunk 4]
```diff
@@ -349,6 +349,15 @@ objects:
...
                 done
+                if ! whoami &> /dev/null; then
+                  if [ -w /etc/passwd ]; then
+                    echo "${USER_NAME:-default}:x:$(id -u):0:${USER_NAME:-default} user:${HOME}:/sbin/nologin" >> /etc/passwd
+                  fi
+                fi
+                eval $(ssh-agent)
+                ssh-add /etc/openshift-installer/ssh-privatekey
+                ssh -A -o PreferredAuthentications=publickey -o StrictHostKeyChecking=false -o UserKnownHostsFile=/dev/null core@${bootstrap_ip} /bin/bash -x /usr/local/bin/installer-gather.sh
+                scp -o PreferredAuthentications=publickey -o StrictHostKeyChecking=false -o UserKnownHostsFile=/dev/null core@${bootstrap_ip}:log-bundle.tar.gz /tmp/artifacts/bootstrap-logs.tar.gz
```
Comment (abhinavdahiya) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml]: would `/tmp/artifacts/installer/bootstrap-logs.tar.gz` be a better choice.

**Comment by abhinavdahiya** [comment]
```console
sh-4.2$ bootstrap_ip=$(python -c \
>                   'import sys, json; d=reduce(lambda x,y: dict(x.items() + y.items()), map(lambda x: x["resources"], json.load(sys.stdin)["modules"])); k="aws_instance.bootstrap"; print d[k]["primary"]["attributes"]["public_ip"] if k in d else ""' \
>                   < /tmp/artifacts/installer/terraform.tfstate
>               )
sh-4.2$ whoami
whoami: cannot find name for user ID 1031310000
sh-4.2$ ls -lah /etc/passwd
-rw-rw-r--. 1 root root 630 Mar  6 02:32 /etc/passwd
sh-4.2$ echo "${USER_NAME:-default}:x:$(id -u):0:${USER_NAME:-default} user:${HOME}:/sbin/nologin" >> /etc/passwd
sh-4.2$ whoami
default
sh-4.2$ eval $(ssh-agent)
Agent pid 36
sh-4.2$ ssh-add /etc/openshift-installer/ssh-privatekey
Identity added: /etc/openshift-installer/ssh-privatekey (/etc/openshift-installer/ssh-privatekey)
sh-4.2$ ssh -A -o PreferredAuthentications=publickey -o StrictHostKeyChecking=false -o UserKnownHostsFile=/dev/null core@${bootstrap_ip} /bin/bash -x /usr/local/bin/installer-gather.sh
Could not create directory '/.ssh'.
Warning: Permanently added '54.209.181.157' (ECDSA) to the list of known hosts.
Gathering bootstrap journals ...
+ ARTIFACTS=/tmp/artifacts
+ echo 'Gathering bootstrap journals ...'
+ mkdir -p /tmp/artifacts/bootstrap/journals
+ for service in bootkube openshift kubelet crio
+ journalctl --boot --no-pager --output=short --unit=bootkube
+ for service in bootkube openshift kubelet crio
+ journalctl --boot --no-pager --output=short --unit=openshift
+ for service in bootkube openshift kubelet crio
+ journalctl --boot --no-pager --output=short --unit=kubelet
+ for service in bootkube openshift kubelet crio
+ journalctl --boot --no-pager --output=short --unit=crio
Gathering bootstrap containers ...
+ echo 'Gathering bootstrap containers ...'
+ mkdir -p /tmp/artifacts/bootstrap/containers
+ sudo crictl ps --all --quiet
+ read -r container
++ grep -oP 'Name: \K(.*)'
++ sudo crictl ps -a --id 76d398bdbefdbe43f513e8adb7c4e84b22000c35f02d662fdf03b0204b7e83ea -v
+ container_name=machine-config-server
+ sudo crictl logs 76d398bdbefdbe43f513e8adb7c4e84b22000c35f02d662fdf03b0204b7e83ea
+ sudo crictl inspect 76d398bdbefdbe43f513e8adb7c4e84b22000c35f02d662fdf03b0204b7e83ea
+ read -r container
++ sudo crictl ps -a --id ef2290a9d7b8899dbb35b8894134f6f1f91f318c66c8a326a172857b5314b6bc -v
++ grep -oP 'Name: \K(.*)'
+ container_name=machine-config-controller
+ sudo crictl logs ef2290a9d7b8899dbb35b8894134f6f1f91f318c66c8a326a172857b5314b6bc
+ sudo crictl inspect ef2290a9d7b8899dbb35b8894134f6f1f91f318c66c8a326a172857b5314b6bc
+ read -r container
+ mkdir -p /tmp/artifacts/bootstrap/pods
+ read -r container
+ sudo podman ps --all --quiet
+ sudo podman logs 192cada536d7
+ sudo podman inspect 192cada536d7
+ read -r container
+ sudo podman logs 8b11a7008838
+ sudo podman inspect 8b11a7008838
+ read -r container
+ sudo podman logs 770e4f9df136
+ sudo podman inspect 770e4f9df136
+ read -r container
+ sudo podman logs 9abfd8340668
+ sudo podman inspect 9abfd8340668
+ read -r container
+ sudo podman logs e98b0ea3dd36
+ sudo podman inspect e98b0ea3dd36
+ read -r container
+ sudo podman logs a4047d8c4229
+ sudo podman inspect a4047d8c4229
+ read -r container
+ sudo podman logs a390c27012c8
+ sudo podman inspect a390c27012c8
+ read -r container
+ sudo podman logs d0b1eae518df
+ sudo podman inspect d0b1eae518df
+ read -r container
+ mkdir -p /tmp/artifacts/control-plane /tmp/artifacts/resources
Gathering cluster resources ...
+ echo 'Gathering cluster resources ...'
+ queue resources/nodes.list oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get nodes -o jsonpath --template '{range .items[*]}{.metadata.name}{"\n"}{end}'
+ local TARGET=/tmp/artifacts/resources/nodes.list
+ shift
++ jobs
++ wc -l
+ local LIVE=0
+ [[ 0 -ge 45 ]]
+ [[ -n '' ]]
+ queue resources/masters.list oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get nodes -o jsonpath -l node-role.kubernetes.io/master --template '{range .items[*]}{.metadata.name}{"\n"}{end}'
+ local TARGET=/tmp/artifacts/resources/masters.list
+ shift
+ sudo oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get nodes -o jsonpath --template '{range .items[*]}{.metadata.name}{"\n"}{end}'
++ wc -l
++ jobs
+ local LIVE=1
+ [[ 1 -ge 45 ]]
+ [[ -n '' ]]
+ queue resources/containers oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get pods --all-namespaces --template '{{ range .items }}{{ $name := .metadata.name }}{{ $ns := .metadata.namespace }}{{ range .spec.containers }}-n {{ $ns }} {{ $name }} -c {{ .name }}{{ "\n" }}{{ end }}{{ range .spec.initContainers }}-n {{ $ns }} {{ $name }} -c {{ .name }}{{ "\n" }}{{ end }}{{ end }}'
+ local TARGET=/tmp/artifacts/resources/containers
+ shift
++ wc -l
++ jobs
+ local LIVE=2
+ [[ 2 -ge 45 ]]
+ [[ -n '' ]]
+ queue resources/api-pods oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get pods -l apiserver=true --all-namespaces --template '{{ range .items }}-n {{ .metadata.namespace }} {{ .metadata.name }}{{ "\n" }}{{ end }}'
+ local TARGET=/tmp/artifacts/resources/api-pods
+ shift
+ sudo oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get pods --all-namespaces --template '{{ range .items }}{{ $name := .metadata.name }}{{ $ns := .metadata.namespace}}{{ range .spec.containers }}-n {{ $ns }} {{ $name }} -c {{ .name }}{{ "\n" }}{{ end }}{{ range .spec.initContainers }}-n {{ $ns }} {{ $name }} -c {{ .name }}{{ "\n" }}{{ end }}{{ end }}'
++ jobs
++ wc -l
+ local LIVE=3
+ [[ 3 -ge 45 ]]
+ [[ -n '' ]]
+ queue resources/apiservices.json oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get apiservices -o json
+ local TARGET=/tmp/artifacts/resources/apiservices.json
+ shift
+ sudo oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get nodes -o jsonpath -l node-role.kubernetes.io/master --template '{range .items[*]}{.metadata.name}{"\n"}{end}'
++ wc -l
++ jobs
+ sudo oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get pods -l apiserver=true --all-namespaces --template '{{ range .items }}-n {{ .metadata.namespace }} {{ .metadata.name }}{{ "\n" }}{{ end }}'
+ local LIVE=4
+ [[ 4 -ge 45 ]]
+ [[ -n '' ]]
+ queue resources/clusteroperators.json oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get clusteroperators -o json
+ local TARGET=/tmp/artifacts/resources/clusteroperators.json
+ shift
+ sudo oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get apiservices -o json
++ wc -l
++ jobs
+ local LIVE=5
+ [[ 5 -ge 45 ]]
+ [[ -n '' ]]
+ queue resources/clusterversion.json oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get clusterversion -o json
+ local TARGET=/tmp/artifacts/resources/clusterversion.json
+ shift
+ sudo oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get clusteroperators -o json
++ wc -l
++ jobs
+ local LIVE=6
+ [[ 6 -ge 45 ]]
+ [[ -n '' ]]
+ queue resources/configmaps.json oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get configmaps --all-namespaces -o json
+ local TARGET=/tmp/artifacts/resources/configmaps.json
+ shift
+ sudo oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get clusterversion -o json
++ wc -l
++ jobs
+ local LIVE=7
+ [[ 7 -ge 45 ]]
+ [[ -n '' ]]
+ queue resources/csr.json oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get csr -o json
+ local TARGET=/tmp/artifacts/resources/csr.json
+ shift
+ sudo oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get configmaps --all-namespaces -o json
++ wc -l
++ jobs
+ local LIVE=8
+ [[ 8 -ge 45 ]]
+ [[ -n '' ]]
+ queue resources/endpoints.json oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get endpoints --all-namespaces -o json
+ local TARGET=/tmp/artifacts/resources/endpoints.json
+ shift
+ sudo oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get csr -o json
++ jobs
++ wc -l
+ local LIVE=9
+ [[ 9 -ge 45 ]]
+ [[ -n '' ]]
+ queue resources/events.json oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get events --all-namespaces -o json
+ local TARGET=/tmp/artifacts/resources/events.json
+ shift
+ sudo oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get endpoints --all-namespaces -o json
++ jobs
++ wc -l
+ local LIVE=10
+ [[ 10 -ge 45 ]]
+ [[ -n '' ]]
+ queue resources/kubeapiserver.json oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get kubeapiserver -o json
+ local TARGET=/tmp/artifacts/resources/kubeapiserver.json
+ shift
+ sudo oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get events --all-namespaces -o json
++ jobs
++ wc -l
+ local LIVE=11
+ [[ 11 -ge 45 ]]
+ [[ -n '' ]]
+ queue resources/kubecontrollermanager.json oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get kubecontrollermanager -o json
+ local TARGET=/tmp/artifacts/resources/kubecontrollermanager.json
+ shift
+ sudo oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get kubeapiserver -o json
++ jobs
++ wc -l
+ local LIVE=12
+ [[ 12 -ge 45 ]]
+ [[ -n '' ]]
+ queue resources/machineconfigpools.json oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get machineconfigpools -o json
+ local TARGET=/tmp/artifacts/resources/machineconfigpools.json
+ shift
+ sudo oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get kubecontrollermanager -o json
++ wc -l
++ jobs
+ local LIVE=13
+ [[ 13 -ge 45 ]]
+ [[ -n '' ]]
+ queue resources/machineconfigs.json oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get machineconfigs -o json
+ local TARGET=/tmp/artifacts/resources/machineconfigs.json
+ shift
+ sudo oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get machineconfigpools -o json
++ jobs
++ wc -l
+ local LIVE=14
+ [[ 14 -ge 45 ]]
+ [[ -n '' ]]
+ queue resources/namespaces.json oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get namespaces -o json
+ local TARGET=/tmp/artifacts/resources/namespaces.json
+ shift
+ sudo oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get machineconfigs -o json
++ jobs
++ wc -l
+ local LIVE=15
+ [[ 15 -ge 45 ]]
+ [[ -n '' ]]
+ queue resources/nodes.json oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get nodes -o json
+ local TARGET=/tmp/artifacts/resources/nodes.json
+ shift
+ sudo oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get namespaces -o json
++ jobs
++ wc -l
+ local LIVE=16
+ [[ 16 -ge 45 ]]
+ [[ -n '' ]]
+ queue resources/openshiftapiserver.json oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get openshiftapiserver -o json
+ local TARGET=/tmp/artifacts/resources/openshiftapiserver.json
+ shift
+ sudo oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get nodes -o json
++ wc -l
++ jobs
+ local LIVE=17
+ [[ 17 -ge 45 ]]
+ [[ -n '' ]]
+ queue resources/pods.json oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get pods --all-namespaces -o json
+ local TARGET=/tmp/artifacts/resources/pods.json
+ shift
+ sudo oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get openshiftapiserver -o json
++ wc -l
++ jobs
+ local LIVE=18
+ [[ 18 -ge 45 ]]
+ [[ -n '' ]]
+ queue resources/rolebindings.json oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get rolebindings --all-namespaces -o json
+ local TARGET=/tmp/artifacts/resources/rolebindings.json
+ shift
+ sudo oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get pods --all-namespaces -o json
++ jobs
++ wc -l
+ local LIVE=19
+ [[ 19 -ge 45 ]]
+ [[ -n '' ]]
+ queue resources/roles.json oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get roles --all-namespaces -o json
+ local TARGET=/tmp/artifacts/resources/roles.json
+ shift
+ sudo oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get rolebindings --all-namespaces -o json
++ jobs
++ wc -l
+ local LIVE=20
+ [[ 20 -ge 45 ]]
+ [[ -n '' ]]
+ queue resources/services.json oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get services --all-namespaces -o json
+ local TARGET=/tmp/artifacts/resources/services.json
+ shift
+ sudo oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get roles --all-namespaces -o json
++ jobs
++ wc -l
+ local LIVE=21
+ [[ 21 -ge 45 ]]
+ [[ -n '' ]]
+ FILTER=gzip
+ queue resources/openapi.json.gz oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get --raw /openapi/v2
+ local TARGET=/tmp/artifacts/resources/openapi.json.gz
+ shift
+ sudo oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get services --all-namespaces -o json
++ jobs
++ wc -l
Waiting for logs ...
+ local LIVE=22
+ [[ 22 -ge 45 ]]
+ [[ -n gzip ]]
+ echo 'Waiting for logs ...'
+ wait
+ gzip
+ sudo oc --config=/opt/openshift/auth/kubeconfig --request-timeout=5s get --raw /openapi/v2
error: the server doesn't have a resource type "pods"
error: the server doesn't have a resource type "nodes"
error: the server doesn't have a resource type "pods"
error: the server doesn't have a resource type "nodes"
error: the server doesn't have a resource type "clusteroperators"
error: the server doesn't have a resource type "apiservices"
error: the server doesn't have a resource type "configmaps"
error: the server doesn't have a resource type "clusterversion"
error: the server doesn't have a resource type "csr"
error: the server doesn't have a resource type "endpoints"
error: the server doesn't have a resource type "kubecontrollermanager"
error: the server doesn't have a resource type "kubeapiserver"
error: the server doesn't have a resource type "events"
error: the server doesn't have a resource type "machineconfigpools"
error: the server doesn't have a resource type "nodes"
error: the server doesn't have a resource type "machineconfigs"
error: the server doesn't have a resource type "namespaces"
error: the server doesn't have a resource type "openshiftapiserver"
error: the server doesn't have a resource type "pods"
error: the server doesn't have a resource type "roles"
Error from server (NotFound): the server could not find the requested resource
error: the server doesn't have a resource type "services"
error: the server doesn't have a resource type "rolebindings"
Gather remote logs
+ echo 'Gather remote logs'
+ MASTERS=()
+ export MASTERS
+ '[' 0 -ne 0 ']'
++ stat --printf=%s /tmp/artifacts/resources/masters.list
+ '[' 0 -ne 0 ']'
++ sudo oc --config=/opt/openshift/auth/kubeconfig whoami --show-server
++ grep -oP 'api.\K([a-z\.]*)'
+ DOMAIN=ci
+ mapfile -t MASTERS
++ dig -t SRV _etcd-server-ssl._tcp.ci +short
++ cut -f 4 -d ' '
++ sed 's/.$//'
/usr/local/bin/installer-gather.sh: line 92: $(dig -t SRV "_etcd-server-ssl._tcp.${DOMAIN}" +short | cut -f 4 -d ' ' | sed 's/.$//'): No such file or directory
+ tar cz -C /tmp/artifacts .
Log bundle written to ~/log-bundle.tar.gz
+ echo 'Log bundle written to ~/log-bundle.tar.gz'
sh-4.2$ scp -o PreferredAuthentications=publickey -o StrictHostKeyChecking=false -o UserKnownHostsFile=/dev/null core@${bootstrap_ip}:log-bundle.tar.gz /tmp/artifacts/bootstrap-logs.tar.gz
Could not create directory '/.ssh'.
Warning: Permanently added '54.209.181.157' (ECDSA) to the list of known hosts.
log-bundle.tar.gz                                                                                                                                       100%   26KB 938.4KB/s   00:00
sh-4.2$ ls -lah /tmp/artifacts/
total 28K
drwxrwsrwx. 3 root    1031310000  52 Apr 18 21:09 .
drwxrwxrwt. 1 root    root        61 Apr 18 21:08 ..
-rw-r--r--. 1 default 1031310000 27K Apr 18 21:09 bootstrap-logs.tar.gz
drwxr-sr-x. 4 default 1031310000 199 Apr 18 21:06 installer
```

this looks like this is working. :yay:

File: ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml (hunk 4 above)
Comment (sdodson) [reply ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml]: Sure, i'll change that real quick.

File: ci-operator/templates/openshift/installer/cluster-launch-installer-openstack-e2e.yaml [hunk 5]
```diff
@@ -385,7 +385,15 @@ objects:
                         --key /tmp/artifacts/installer/tls/journal-gatewayd.key \
                         --url "https://${bootstrap_ip}:19531/entries?_SYSTEMD_UNIT=${service}.service"
                 done
-              fi
```
Comment (wking) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-openstack-e2e.yaml]: I think you need to keep a closing `fi` on `if [ -n "${bootstrap_ip}" ]`, although it should live after the block you add below, but before the `else` for `if [ -f /tmp/artifacts/installer/terraform.tfstate ]`.

File: ci-operator/templates/openshift/openshift-ansible/cluster-launch-e2e-40.yaml [hunk 6]
```diff
@@ -387,6 +387,40 @@ objects:
...
+                done
+                if ! whoami &> /dev/null; then
+                  if [ -w /etc/passwd ]; then
+                    echo "${USER_NAME:-default}:x:$(id -u):0:${USER_NAME:-default} user:${HOME}:/sbin/nologin" >> /etc/passwd
+                  fi
+                fi
+                eval $(ssh-agent)
+                ssh-add /etc/openshift-installer/ssh-privatekey
+                ssh -A -o PreferredAuthentications=publickey -o StrictHostKeyChecking=false -o UserKnownHostsFile=/dev/null core@${bootstrap_ip} /bin/bash -x /usr/local/bin/installer-gather.sh
+                scp -o PreferredAuthentications=publickey -o StrictHostKeyChecking=false -o UserKnownHostsFile=/dev/null core@${bootstrap_ip}:log-bundle.tar.gz /tmp/artifacts/installer/bootstrap-logs.tar.gz              fi
```
Comment (wking) [inline ci-operator/templates/openshift/openshift-ansible/cluster-launch-e2e-40.yaml]: ah, also we want to drop the `fi` here.

File: ci-operator/templates/openshift/installer/cluster-launch-installer-src.yaml [hunk 7]
```diff
@@ -367,6 +367,15 @@ objects:
...
                 done
+                if ! whoami &> /dev/null; then
+                  if [ -w /etc/passwd ]; then
+                    echo "${USER_NAME:-default}:x:$(id -u):0:${USER_NAME:-default} user:${HOME}:/sbin/nologin" >> /etc/passwd
+                  fi
+                fi
+                eval $(ssh-agent)
+                ssh-add /etc/openshift-installer/ssh-privatekey
+                ssh -A -o PreferredAuthentications=publickey -o StrictHostKeyChecking=false -o UserKnownHostsFile=/dev/null core@${bootstrap_ip} /bin/bash -x /usr/local/bin/installer-gather.sh
+                scp -o PreferredAuthentications=publickey -o StrictHostKeyChecking=false -o UserKnownHostsFile=/dev/null core@${bootstrap_ip}:log-bundle.tar.gz /tmp/artifacts/installer/bootstrap-logs.tar.gz              fi
```
Comment (wking) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-src.yaml]: Still have a trailing `fi` here.

File: ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml (hunk 2 above)
Comment (sdodson) [reply ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml]: made writable in https://github.com/openshift/origin/pull/22592

**Comment by wking** [comment]
/lgtm

Dunno what's up with CI, though.

**Pull request merged**